mobile/library/java/io/envoyproxy/envoymobile/engine/EnvoyFinalStreamIntelImpl.java

```diff
@@ -2,7 +2,8 @@
 
 import io.envoyproxy.envoymobile.engine.types.EnvoyFinalStreamIntel;
 
-class EnvoyFinalStreamIntelImpl implements EnvoyFinalStreamIntel {
+// `public` for testing only.
+public class EnvoyFinalStreamIntelImpl implements EnvoyFinalStreamIntel {
```

Comment:
You can use `@VisibleForTesting` annotation similar to https://github.com/envoyproxy/envoy/blob/29d30aa2ea190d67c04cdc26f23c5c9954460ab4/mobile/library/java/io/envoyproxy/envoymobile/engine/AndroidNetworkMonitor.java#L19

Reply:
good suggestion thanks!